src/dispatch.rs

```diff
@@ -543,9 +543,9 @@ pub(crate) fn gen_dispatch_str(cmd: DispatchType, dispatch: bool) -> crate::Resu
         MoveOutOfGroup => "moveoutofgroup".to_string(),
     };
     if let SetCursor(_, _) = cmd {
-        Ok(format!("setcursor {string_to_pass}"))
+        Ok(format!("/setcursor {string_to_pass}"))
```

Comment:
Can you instead try adding (and testing, not sure if this breaks all of cmds or not) it to the `write_to_socket` function?

Reply:
I try but it doesn't works, in autoname it crash instant on getting Clients.

```
diff --git a/src/dispatch.rs b/src/dispatch.rs
index e7e5bf0..88d4ffb 100644
--- a/src/dispatch.rs
+++ b/src/dispatch.rs
@@ -546,9 +546,9 @@ pub(crate) fn gen_dispatch_str(cmd: DispatchType, dispatch: bool) -> crate::Resu
         MoveOutOfGroup => "moveoutofgroup".to_string(),
     };
     if let SetCursor(_, _) = cmd {
-        Ok(format!("/setcursor {string_to_pass}"))
+        Ok(format!("setcursor {string_to_pass}"))
     } else if dispatch {
-        Ok(format!("/dispatch {string_to_pass}"))
+        Ok(format!("dispatch {string_to_pass}"))
     } else {
         Ok(string_to_pass)
     }
diff --git a/src/shared.rs b/src/shared.rs
index 8f30b38..6fb1c3a 100644
--- a/src/shared.rs
+++ b/src/shared.rs
@@ -194,7 +194,11 @@ pub(crate) async fn write_to_socket(path: String, content: &[u8]) -> crate::Resu
 
     let mut stream = UnixStream::connect(path).await?;
 
-    stream.write_all(content).await?;
+    let content_with_slash: Vec<u8> = std::iter::once(b'/')
+        .chain(content.iter().copied())
+        .collect();
+
+    stream.write_all(&content_with_slash).await?;
 
     let mut response = vec![];
 
@@ -218,7 +222,11 @@ pub(crate) fn write_to_socket_sync(path: String, content: &[u8]) -> crate::Resul
     use std::os::unix::net::UnixStream;
     let mut stream = UnixStream::connect(path)?;
 
-    stream.write_all(content)?;
+    let content_with_slash: Vec<u8> = std::iter::once(b'/')
+        .chain(content.iter().copied())
+        .collect();
+
+    stream.write_all(&content_with_slash)?;
 
     let mut response = vec![];
 
```

Reply:
There's probably a bigger refactoring needed, but it's a good idea @yavko!

In general, the `/` separates flags from command, and in some cases this project does make use of it:

https://github.com/hyprland-community/hyprland-rs/blob/8764ec29bdd26dc88f37b6e53364b4e7ba06994c/src/keyword.rs#L100-L103

https://github.com/hyprland-community/hyprland-rs/blob/8764ec29bdd26dc88f37b6e53364b4e7ba06994c/src/data/regular.rs#L12

One idea to handle this could perhaps be to make `write_to_socket` accept `flags` (e.g. `""` or `"j"`) and `data` as separate arguments, and let it combine the two using `/` char.

This way, there would be exactly one place where the slash would be added, and thus hopefully an easier to follow and to maintain.

I don't know if you'd prefer to merge this PR as is first (as it does solve the very specific issue with `dispatch` for users of @cyrinux's project) and do the refactoring afterwards, or whether you prefer to work on a proper fix right away?

Reply:
I agree that a flag would be needed, but I don't like the idea of using a string, instead could we use a enum? Also I would like to keep this in here for the time being as I prefer universal fixes.